extensions/amp-animation/0.1/web-animation-types.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @typedef {!WebMultiAnimationDef|!WebKeyframeAnimationDef}
+ */
+export let WebAnimationDef;
+
+
+/**
+ * @mixes WebAnimationTimingDef
+ * @typedef {{
+ *   animations: !Array<!WebAnimationDef>,
```

Comment:
Also, another nuance, I wanted all of the following to be valid configs:

Shortest key-frame only config:
```
{
  target: target1,
  duration: 1000,
  keyframes: {opacity: 0}
}
```

An array of animations:
```
[
  {
    target: target1,
    duration: 1000,
    keyframes: {opacity: 1}
  },
  {
    target: target2,
    duration: 500,
    delay: 500,
    keyframes: {opacity: 1}
  }
]
```

And an object + array:
```
{
  duration: 1000,
  animations: [
    {
      target: target1,
      keyframes: {opacity: 1}
    },
    {
      target: target2,
      duration: 500,
      delay: 500,
      keyframes: {opacity: 1}
    }
  ]
}
```

WDYT?